cluster/crypto/Makefile

```diff
@@ -0,0 +1,56 @@
+
+.PHONY: all clean
+all: all.tar
+
+clean:
+	rm -f *.tar
+	rm -f *.pem
+	rm -f *.csr
+	rm -f ca.srl
+	rm -f openssl.cnf
+
+# config
+openssl.cnf: openssl.cnf.in san-extras
+	rm -f openssl.cnf
+	cat openssl.cnf.in >> openssl.cnf
+	cat san-extras >> openssl.cnf
+
+# root CA
+ca-key.pem:
```

Comment:
I'd like to colapse all csr, publickey, and tar rules into:
```makefile
%.csr: %-key.pem
	openssl req -new -key "$<" -out "$@" -subj "/CN=kubelet"

%.pem: %.csr ca.pem ca-key.pem
	openssl x509 -req -in "$<" -CA ca.pem -CAkey ca-key.pem -CAcreateserial -out "$@" -days 365

%.tar: %-key.pem %.pem
	tar cfv "$@" $^
```
but appearantly you can't match a wildcard rule with another wildcard rule. I am no make expert though